### DISCUSSION TITLE
_split_datalistAPI

### DISCUSSION THREAD

**Discussion by gkrisp98** [Project-MONAI/MONAI#4669] [Q&A]
Hi, I am trying to use Cross Validation from MONAI on a nifti dataset. I have created the dataset like this:
`dataset = ImageDataset(image_files = images, labels = labels)` and I define the Cross Validation:
`cvdataset = CrossValidation(
    dataset_cls=dataset,
    nfolds=5,
    seed=12345,
    section="training",
    transform=train_transform,
)`
When I try to run it to get the partitions I am getting this error: dataset class must have _split_datalist API.
Can someone help me to get over this error and use CV ?

**Comment by JohnMasoner** [comment]
Hey @gkrisp98 ,

The following code is the location of the error that prompted you:

```python
if not hasattr(dataset_cls, "_split_datalist"):
    raise ValueError("dataset class must have _split_datalist API.")
```

The mean the `ImageDataset` have no  `_split_datalist`.

you could create the dataset by `DecathlonDataset` it covers `_split_datalist`.

Hope this is useful to you.
